### PR TITLE
Development Environment Configuration

### PR DESCRIPTION
**Pull request by tpendragon**
1. Fix Solr configuration to use 7.x locally (like CircleCI)
2. Add documentation on dependencies to run tests locally.
3. Fix periodically failing spec.
4. Add local coverage generation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,6 +213,30 @@ Instead, each directive may contain these arguments:
   * `:compression` (Integer) : The left number of the compression ratio `n:1`, e.g. 12 will apply 12:1 compression. Default: 10.
   * `:tile_size` (Integer) : Pixel dimension of the tiles. Default: 1024
 
+# Development Environment
+
+## Dependencies
+
+* [ImageMagick](https://www.imagemagick.org)
+    * On a mac, do `brew install imagemagick --with-little-cms2 --with-openjpeg`
```
Comment (botimer) [inline README.md]: As of [Homebrew 2.0.0](https://brew.sh/2019/02/02/homebrew-2.0.0/), homebrew-core accepts no options and uses both of these by default.

File: spec/units/derivatives_spec.rb [hunk 2]
```diff
@@ -40,6 +40,12 @@ class CustomFile < ActiveFedora::Base
       end
     end
     describe "#{method}=" do
+      around do |obj|
```
Comment (botimer) [inline spec/units/derivatives_spec.rb]: Isn't this save/restore usually done as follows?

```
around(:each) do |example|
  ..
  example.run
  ..
end
```

**Comment by botimer** [comment]
Following the updated README (without imagemagick options) gave me a fully passing suite. Once that's addressed, 🚢.

**Comment by tpendragon** [comment]
@botimer Sorry, this should be ready now. Was lost in coverage fixing.

File: spec/units/derivatives_spec.rb (hunk 2 above)
Comment (tpendragon) [reply spec/units/derivatives_spec.rb]: `example.run` is definitely better. Rubocop complains about `(:each)`.

**Pull request merged**